_playwright-tests/UI/SnapshotRepo.spec.ts

```diff
@@ -102,4 +103,50 @@ test.describe('Snapshot Repositories', () => {
       await expect(row).not.toBeVisible();
     });
   });
+
+  test('test snapshot manual trigger', async ({ page }) => {
+    await navigateToRepositories(page);
+    await closePopupsIfExist(page);
+    const repoNamePrefix = 'snapshot-manual-trigger';
+    const randomName = () => `${(Math.random() + 1).toString(36).substring(2, 6)}`;
+    const repoName = `${repoNamePrefix}-${randomName()}`;
+    const url = randomUrl();
+    await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+
+    await test.step('Create a repository', async () => {
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.getByRole('dialog', { name: 'Add custom repositories' })).toBeVisible();
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('Introspect only').click();
+      await page.getByLabel('URL').fill(url);
+      await page.getByRole('button', { name: 'Save', exact: true }).click();
+      // Need to wait for introspection task to finish
+      await page.waitForTimeout(7000);
+      const row = await getRowByNameOrUrl(page, repoName);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+
+    await test.step('Edit the repository and trigger snapshot manually', async () => {
+      const row = await getRowByNameOrUrl(page, repoName);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+      await row.getByLabel('Kebab toggle').click();
+      await row.getByRole('menuitem', { name: 'Edit' }).click({ timeout: 60000 });
+      await page.getByLabel('Name').fill(`${repoName}-edited`);
+      await page.getByLabel('Snapshotting').click();
+      await page.getByRole('button', { name: 'Save changes', exact: true }).click();
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+      // Need to wait for the introspection task to finish
+      await page.waitForTimeout(20000);
+      await row.getByLabel('Kebab toggle').click();
+      // Trigger a snapshot manually
+      await row.getByRole('menuitem', { name: 'Trigger snapshot' }).click();
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+      await row.getByLabel('Kebab toggle').click();
+      await page.getByRole('menuitem', { name: 'View all snapshots' }).click();
+      // asssert that triggered snapshot is in the list
+      await expect(page.getByLabel('snapshot list table').locator('tbody')).toBeVisible();
+      await page.getByTestId('snapshot_list_modal-ModalBoxCloseButton').click();
```

Comment:
this doesn't really assert what the commit says no? 🤔 
and is there no better locator for that close button? 🔍

Reply:
Yes, there is a solution, but it specifically targets the 'cross' and 'close' buttons within the modal. Adding first().click() does work; however, I was trying to avoid using first()
```     
 await page.getByRole('button', { name: 'Close', exact: true }).first().click();
```